### PR TITLE
add fp8 scaled_mm for XPU

### PR DESCRIPTION
**Pull request by yuchengliu1**
add fp8 scaled_mm for XPU
Fixes #ISSUE_NUMBER

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @gujinghui @PenghuiCheng @jianyuh @min-jean-cho @yanbing-j @Guobing-Chen @Xia-Weiwen @snadampal @voznesenskym @penguinwu @EikanWang @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

**Comment by Xia-Weiwen** [comment]
CC @Guobing-Chen and @yanbing-j as this PR tries to change ideep version.

**Comment by yanbing-j** [comment]
> CC @Guobing-Chen and @yanbing-j as this PR tries to change ideep version.

I suppose this is a draft PR to involve oneDNN v3.6 inside. @yuchengliu1 Could you please use ghstack to split the ideep/onednn change and other code change? And after we upgrade oneDNN into PyTorch, you can rebase this ghstack.

**Review by mingfeima — CHANGES_REQUESTED** [review]
this PR right now has a lot of unnecessary format changes, can we just remove such changes. Usually if we want to scrap format changes, we can open a separate PR to do the job, otherwise it is harder to tell what this PR is really about.

File: test/inductor/test_fp8.py [hunk 1]
```diff
@@ -112,19 +111,22 @@ def f(x):
     @unittest.skipIf(not PLATFORM_SUPPORTS_FP8, f8_msg)
```
Comment (airMeng) [inline test/inductor/test_fp8.py:111]: remember to update the definition of this

**Comment by EikanWang** [comment]
It does not make sense to change non-XPU related code. Pls. refine this PR. Meanwhile, pls. ensure your PR is small enough for the sake of review.

**Review by EikanWang — CHANGES_REQUESTED** [review]
Pls. refine the PR and attach right labels!

**Pull request closed without merging**